src/develop/pixelpipe_hb.c

```diff
@@ -2880,13 +2891,30 @@ float *dt_dev_get_raster_mask(const dt_dev_pixelpipe_t *pipe,
   float *raster_mask = NULL;
 
   GList *source_iter;
-  for(source_iter = pipe->nodes; source_iter; source_iter = g_list_next(source_iter))
+  for(source_iter = piece->pipe->nodes; source_iter; source_iter = g_list_next(source_iter))
   {
     const dt_dev_pixelpipe_iop_t *candidate = (dt_dev_pixelpipe_iop_t *)source_iter->data;
+
+    if((candidate->module == target_module)
+       || (candidate->module->iop_order >= target_module->iop_order))
+    {
+      dt_control_log
+        (_("module `%s' can't get raster mask from module `%s'"
+           "as that is processed later in the pixel pipe.\n"
+           "raster mask is ignored."),
+           target_module->so->op, raster_mask_source->so->op);
```

Comment:
I'll update this to use the module name and not the file-name. That is, using target_module->name() which is also translated. Also adding a space before 'as'.